### PR TITLE
log events in the interlocking component

### LINKED ISSUE #418: Log Events in the interlocking
## User stories
As a logger, I want certain events to be logged in order to provide information about those events.

## (Wiki) References
- [List of availabe logging functions](https://github.com/BP2022-AP1/bp2022-ap1/wiki/Logging#log-creator)

## Acceptance Criteria
- [x] log whenever a fahrstrasse is created
- [x] log whenever a fahrstrasse is removed
- [x] log when a signal is set by the infrastructure provider
- [x] log when a train enters a block section
- [x] log when a train leaves a block section

### PR DESCRIPTION
**Pull request by Lietze**
Fixes #418

Description (what might a PO or reviewer want to know?)

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [x] Docs (code, wiki & diagrams) updated
- [x] Breaking changes anounced
- [x] Added classes that inherit from `BaseModel` to `src.constants.tables`
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

File: src/interlocking_component/infrastructure_provider.py [hunk 1]
```diff
@@ -64,8 +80,9 @@ def train_drove_onto_track(self, train: Train, edge: Edge):
         self.tds_count_in(track_segment_id)
 
         self.route_controller.maybe_set_fahrstrasse(train, edge)
+        print(train.identifier)
```
Comment (LucasDerReisende) [inline src/interlocking_component/infrastructure_provider.py]: maybe remove this print if not needed

**Review by LucasDerReisende — APPROVED** [review]
looks good :)

**Comment by LucasDerReisende** [comment]
I am just wondering, why the coverage decreased here by that much?

File: src/interlocking_component/infrastructure_provider.py (hunk 1 above)
Comment (CR1337) [inline src/interlocking_component/infrastructure_provider.py]: Maybe you want to remove this?

File: src/interlocking_component/route_controller.py [hunk 2]
```diff
@@ -194,7 +196,14 @@ def set_fahrstrasse(self, train: Train, edge: Edge):
         new_route = self.router.get_route(edge, train.timetable[0].edge)
         # new_route contains a list of signals from starting signal to end signal of the new route.
 
-        for end_node_candidat in new_route[1:]:
+        route_length = 0
+
+        for i in range(new_route - 1):
```
Comment (CR1337) [inline src/interlocking_component/route_controller.py]: `new_route` is a list of nodes, right?
Then this doesn't work. `range` uses an integer and not an iterable.
Either use `for node in new_route` or if you really need an index use `for i in range(len(new_route))`. 
If you need both use `for i, node in enumerate(new_route)`.

**Review by CR1337 — CHANGES_REQUESTED** [review]
Overall it looks good. Please just have a quick look at my comments :)

File: src/interlocking_component/infrastructure_provider.py (hunk 1 above)
Comment (Lietze) [reply src/interlocking_component/infrastructure_provider.py]: Yes, that slipped through.

Comment (Lietze) [reply src/interlocking_component/infrastructure_provider.py]: Yes I did

File: src/interlocking_component/route_controller.py (hunk 2 above)
Comment (Lietze) [reply src/interlocking_component/route_controller.py]: I changed this

**Review by CR1337 — APPROVED** [review]
:+1:

**Comment by Lietze** [comment]
> I am just wondering, why the coverage decreased here by that much?

The only thing I can think about is, that I am now using a MockLogger instead of the real one. Maybe I was testing some code of the logger before? If so, this would not be the place to test that code

**Comment by LucasDerReisende** [comment]
It seems like it's not the Logger, but in `route_controller.py` and `simulation_objects.py`. Could you please have a look at those, because when looking at the coverage difference on dev, it seems like it decreased by 0.4%. If you know the reason and it's intended, I don't think it's a problem, just want to make sure, that we did not remove tests for certain functions.

**Pull request merged**